0016-base-primitive-unification.md

```diff
@@ -0,0 +1,115 @@
+# Base Primitive and Units of Primitive Work
+
+| **Status**        | **Proposed/Accepted/Deprecated**             |
+|:------------------|:---------------------------------------------|
+| **RFC #**         | 0016                                         |
+| **Authors**       | Lev Bishop (lsbishop@us.ibm.com)             |
+|                   | Ian Hincks (ian.hincks@ibm.com)              |
+|                   | Blake Johnson (blake.johnson@ibm.com)        |
+|                   | Chris Wood (cjwood@us.ibm.com)               |
+| **Submitted**     | 2023-10-26                                   |
+| **Updated**       | YYYY-MM-DD     
+
+## Summary
+
+The primitives form an execution framework for QPUs and their simulators.
+Each primitive specializes at some computational task.
+For example, the `Estimator` specializes at computing expectation values of user-provider observables.
+
+Different types of primitives are implemented as different abstract subclasses of `BasePrimitive` (e.g. `BaseEstimator`), and different primitive implementations as further subclasses (e.g. `qiskit_aer.primitives.Estimator`).
+Because the constructor of these subclasses is reserved for implementation-specific setup, the main entry-point to execution is the `.run()` method of a particular class.
+However, as it stands, `BasePrimitive` does not provide `.run()` as an abstraction; each type of primitive gets to define its own signature.
+
+This RFC proposes an abstraction for `BasePrimitive.run()` as well as structures for its inputs and outputs.
+
+## Motivation
+
+The central motivation of this RFC is to more clearly emphasize that primitives are a well-defined execution framework on top of which applications can be built.
+We aim to force each category of primitive to clearly state what units of quantum work (tasks) it is able to perform, and what outputs can be expected.
+It is understood that having the proposed abstract base method alone will not necessarily improve the day-to-day life of a typical user of the primitives, and neither will it necessarily improve the ease of implementing new primitive types or implementations: this change is mainly about clarifying the shared nature of the primitives, and reflecting this view in abstractions.
```

Comment:
I'm not sure I fully understand this. The motivation here is just to have a common subclass and to say that all primitives have a `run()` method via that common parent?